### PR TITLE
[Distributed] Fix availability of DAS requirements and always emit into client whenLocal with typed throws

### PR DESCRIPTION
**Pull request by ktoso**
When adopting typed throws in `DistributedActor/whenLocal` we missed marking it as always emit into client, which is neccessary for compatibility reasons. The old "rethrows" version of this method exists for ABI reasons so that's not an issue, but the new func must be AEIC.

Related to https://github.com/apple/swift/issues/74291
Part of rdar://129651474

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: stdlib/public/Distributed/DistributedActor.swift [hunk 1]
```diff
@@ -362,6 +362,8 @@ extension DistributedActor {
   /// state.
   ///
   /// When the actor is remote, the closure won't be executed and this function will return nil.
+  @inlinable
+  @_alwaysEmitIntoClient
```
Comment (Azoy) [inline stdlib/public/Distributed/DistributedActor.swift:366]: AEIC implies `@inlinable`.

Comment (ktoso) [reply stdlib/public/Distributed/DistributedActor.swift:366]: Maybe so... All sites where I see typed throws adopted spell it out doubly-so like this... I'll follow the pattern and we can do repo wide cleanup pass separately perhaps

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Pull request merged**